lib/avro-patches/logical_types/io.rb

```diff
@@ -2,7 +2,7 @@
   def write_data(writers_schema, logical_datum, encoder)
     datum = writers_schema.type_adapter.encode(logical_datum)
 
-    unless Avro::Schema.validate(writers_schema, datum, false, true)
+    unless Avro::Schema.validate(writers_schema, datum, { recursive: false, encoded: true })
```

Comment:
Yep, I believe that works. Can't actually test it in the context of this gem due to the use of `prepend` :man_facepalming: